### PR TITLE
Use page instead of demo

### PR DESCRIPTION
**Pull request by balloob**
## Proposed change

Use "page" instead of "demo" now that we're adding markdown to each page.

Also adds an editing page.

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: build-scripts/gulp/gallery.js [hunk 1]
```diff
@@ -19,32 +19,32 @@ require("./service-worker.js");
...
     }
-    processed.add(demoId);
+    processed.add(pageId);
 
-    const [category, name] = demoId.split("/", 2);
+    const [category, name] = pageId.split("/", 2);
 
-    const demoFile = path.resolve(demoDir, `${demoId}.ts`);
-    const descriptionFile = path.resolve(demoDir, `${demoId}.markdown`);
+    const demoFile = path.resolve(pageDir, `${pageId}.ts`);
```
Comment (zsarnett) [inline build-scripts/gulp/gallery.js:46]:
```suggestion
    const pageFile = path.resolve(pageDir, `${pageId}.ts`);
```

File: build-scripts/gulp/gallery.js [hunk 2]
```diff
@@ -19,32 +19,32 @@ require("./service-worker.js");
...
+    const descriptionFile = path.resolve(pageDir, `${pageId}.markdown`);
     const hasDemo = fs.existsSync(demoFile);
```
Comment (zsarnett) [inline build-scripts/gulp/gallery.js:48]:
```suggestion
    const hasDemo = fs.existsSync(pageFile);
```

File: build-scripts/gulp/gallery.js [hunk 3]
```diff
@@ -93,51 +94,54 @@ gulp.task("gather-gallery-demos", async function gatherDemos() {
...
+        group.pages = [];
+      }
       continue;
     }
 
     // Any pre-defined groups will not be sorted.
-    if (group.demos) {
-      for (const demo of group.demos) {
+    if (group.pages) {
+      for (const demo of group.pages) {
```
Comment (zsarnett) [inline build-scripts/gulp/gallery.js:120]:
```suggestion
      for (const page of group.pages) {
```

File: build-scripts/gulp/gallery.js [hunk 4]
```diff
@@ -93,51 +94,54 @@ gulp.task("gather-gallery-demos", async function gatherDemos() {
...
         if (!toProcess.delete(demo)) {
           console.error("Found unreferenced demo", demo);
```
Comment (zsarnett) [inline build-scripts/gulp/gallery.js:122]:
```suggestion
        if (!toProcess.delete(page)) {
          console.error("Found unreferenced page", page);
```

**Review by zsarnett — DISMISSED** [review]

File: build-scripts/gulp/gallery.js (hunk 1 above)
Comment (balloob) [reply build-scripts/gulp/gallery.js:46]: each page contains a demo and a description. The ts file is the demo.

**Pull request merged**